### PR TITLE
remove taxonnotfoundexception for real

### PR DESCRIPTION
**Pull request by southeo**
In case a taxon of interest is not in the database, the program checks the list of provided taxon_ids to the list of taxon_ids actually retrieved from the database. If there are taxa in the provided list that were not in the retrieved taxon_ids, they go into an "unknown" list and are returned with unknown confidence.

The issue comes from: the provided_taxon_id list was a list of strings and the list from the database was a list of ints, so everything was going into the "unknown list". A quick conversion fixed this.

### CODE REVIEW

File: flask_app/api/predictions.py [hunk 1]
```diff
@@ -262,32 +262,30 @@ def resolve_taxa(top_taxa_ids, confidence):
...
 
 
-
 def append_missed_taxa(missed_taxa, taxa):
     for tid in missed_taxa:
         d = {
-            "taxon_id":tid,
-            "confidence":"Unknown"
+            "taxon_id": tid,
+            "confidence": "Unknown"
```
Comment (TomDijkema) [inline flask_app/api/predictions.py:272]: Clean :)

File: flask_app/api/predictions.py [hunk 2]
```diff
@@ -312,8 +310,9 @@ def controller(relation, taxon_id, is_subject_query, conf_thresh, strict, check_
         species_list, taxon_ids = get_species_list_from_phyla(appropriate_phyla)
         missed_taxa = []
     else:
+        check_list = [eval(i) for i in check_list]
         species_list, taxon_ids = get_taxonomy_from_species_list(check_list)
-        missed_taxa = list_diff(check_list, taxon_ids) # this is the list of taxa not in the database
+        missed_taxa = list_diff(check_list, taxon_ids)  # this is the list of taxa not in the database
```
Comment (TomDijkema) [inline flask_app/api/predictions.py]: Good that you found out why it was not working! If I may, I would recommend to use some other method than **eval**. Eval is considered not the best to make use of: https://stackoverflow.com/questions/1832940/why-is-using-eval-a-bad-practice

**Review by TomDijkema — COMMENTED** [review]
Very happy this is solved! Please look at the single comment about eval whereafter I will approve :)

File: flask_app/api/predictions.py (hunk 2 above)
Comment (southeo) [reply flask_app/api/predictions.py]: Did not know that! Thanks for the heads-up :)

**Review by samleeflang — APPROVED** [review]

**Pull request merged**